src/plat/spike/machine/hardware.c

```diff
@@ -62,8 +73,7 @@ interrupt_t
 getActiveIRQ(void)
 {
 
-    uint64_t temp = 0;
-    asm volatile ("csrr %0, scause":"=r" (temp)::);
+    word_t temp = read_csr_env(cause);
```

Comment:
You had a previous commit that "Replace S-mode instructions/registers with privilege-aware definitions", should this not also be in there?

Reply:
I separate commits/changes in the architecture (starting with "riscv/arch:"), from changes in the platform ("spike:")